src/olympia/lib/settings_base.py

```diff
@@ -1496,15 +1496,39 @@ def get_sentry_release():
     return version
 
 
-# This is the DSN to the Sentry service.
+def scrub_sensitive_data(event, hint):
+    def _scrub_sensitive_data_recursively(data, name=None):
+        # This only works with lists or dicts but we shouldn't need anything else.
+        if isinstance(data, (list, dict)):
+            items = data.items() if isinstance(data, dict) else enumerate(data)
+            for key, value in items:
+                data[key] = _scrub_sensitive_data_recursively(value, name=key)
```

Comment:
mmm, for `list`s `key` would just be the list index integer, which is never going to match `SENTRY_SENSITIVE_FIELDS`. So should we be passing down the list name instead?  

Example:
```python
{"email": ["a@foo.com", "b@baa.com"]}
```
In the code above wouldn't redact any of the email addresses because `name` would be `0` and `1`.

Reply:
To be fair if we had something like that we'd likely call the key `emails` and it would never match `SENTRY_SENSITIVE_FIELDS` in its current state either :stuck_out_tongue: 

I think the logic breaks down quite quickly for cases like that and it's always going to be a best effort kind of thing. If we had `{'email': [{'value': 'a@foo.com'}, {'value': 'b@foo.com'}]}` for instance what should we do...

Reply:
We could make the match to `SENTRY_SENSITIVE_FIELDS` "sticky" and redact all values under it... that may be a bit too extreme though (and very prone to false positives)